packages/synapse-react-client/src/components/SynapseChat/SynapseChat.tsx

```diff
@@ -0,0 +1,131 @@
+  const { mutate: createAgentSession } = useCreateAgentSession({
+    onSuccess: newAgentSession => {
+      setAgentSessionId(newAgentSession.sessionId)
+    },
+  })
+  const theme = useTheme()
+  const [messages, setMessages] = useState([initialMessage])
+  const [currentMessage, setCurrentMessage] = useState('')
+
+  useEffect(() => {
+    // on mount, create a new agent session!
+    if (createAgentSession) {
+      createAgentSession({
+        agentAccessLevel: AgentAccessLevel.PUBLICLY_ACCESSIBLE,
+        agentId: agentId,
+      })
+    }
+  }, [createAgentSession])
+
+  const handleSendMessage = () => {
+    if (currentMessage.trim()) {
+      setMessages([...messages, currentMessage])
+      setCurrentMessage('')
+    }
+  }
+
+  const handleKeyDown: KeyboardEventHandler<
+    HTMLInputElement | HTMLTextAreaElement
+  > = event => {
+    if (event.key === 'Enter') {
+      event.preventDefault()
+      handleSendMessage()
+    }
+  }
+  const isMessage = !!currentMessage
+  const sendMessageButtonColor = (theme.palette.secondary as ColorPartial)[300]
+  if (!agentSessionId) {
+    return <SkeletonParagraph numRows={6} />
+  }
+
+  return (
+    <Box
+      display="flex"
+      flexDirection="column"
+      justifyContent="space-between"
+      height="100vh"
+      maxWidth="700px"
+      mx="auto"
+      p={2}
+    >
+      <Paper elevation={3} sx={{ flexGrow: 1, overflowY: 'auto', p: 2, mb: 2 }}>
+        <Typography variant="h6" gutterBottom>
+          SynapseChat
+        </Typography>
+        <List>
+          {messages.map((message, index) => {
+            return (
+              <SynapseChatInteraction
+                key={index}
+                sessionId={agentSessionId}
+                userMessage={message}
+              />
+            )
+          })}
+        </List>
+      </Paper>
+      <Box component="form" onSubmit={handleSendMessage}>
+        <FormControl fullWidth sx={{ m: 1 }}>
+          <OutlinedInput
+            value={currentMessage}
+            endAdornment={
+              <InputAdornment
+                position="end"
+                sx={{ p: '0px', ml: '7px', mr: '13px' }}
+              >
+                <IconButton
+                  disabled={!isMessage}
+                  onClick={handleSendMessage}
+                  sx={{
+                    color: sendMessageButtonColor,
+                    borderStyle: 'solid',
+                    borderWidth: isMessage ? '2px' : '1px',
+                    borderColor: isMessage ? sendMessageButtonColor : 'gray',
+                  }}
+                >
+                  <ArrowUpward />
+                </IconButton>
+              </InputAdornment>
+            }
+            placeholder={'Message SynapseChat'}
+            sx={{ borderRadius: 96.6 }}
+            onChange={e => setCurrentMessage(e.target.value)}
+            onKeyDown={handleKeyDown}
+          />
+        </FormControl>
```

Comment:
`TextField` can do all of this, I think this should be pretty close to what you have:

```suggestion
        <TextField
          fullWidth
          value={currentMessage}
          onChange={e => setCurrentMessage(e.target.value)}
          onKeyDown={handleKeyDown}
          placeholder={'Message SynapseChat'}
          sx={{ m: 1 }}
          InputProps={{
            sx: { borderRadius: 96.6 },
            endAdornment: (
              <IconButton
                disabled={!isMessage}
                onClick={handleSendMessage}
                sx={{
                  ml: '7px',
                  mr: '13px',
                  color: sendMessageButtonColor,
                  borderStyle: 'solid',
                  borderWidth: isMessage ? '2px' : '1px',
                  borderColor: isMessage ? sendMessageButtonColor : 'gray',
                }}
              >
                <ArrowUpward />
              </IconButton>
            ),
          }}
        />
```

Reply:
That works for me!